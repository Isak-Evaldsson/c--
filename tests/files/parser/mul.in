int a() {
    // Both expressions shall be equivalent
    int a = 3 - 2 * 2 / 6;
    int b = 3 - ((2 * 2) / 6);

    // However this shall not
    int c = (3 - 2) * (2 / 6);
}