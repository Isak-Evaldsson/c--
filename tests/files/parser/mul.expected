Prototype: a
    VarDecl: a
        BinOpExpr: -
            LiteralExpr: 3
            BinOpExpr: /
                BinOpExpr: *
                    LiteralExpr: 2
                    LiteralExpr: 2
                LiteralExpr: 6
    VarDecl: b
        BinOpExpr: -
            LiteralExpr: 3
            BinOpExpr: /
                BinOpExpr: *
                    LiteralExpr: 2
                    LiteralExpr: 2
                LiteralExpr: 6
    VarDecl: c
        BinOpExpr: *
            BinOpExpr: -
                LiteralExpr: 3
                LiteralExpr: 2
            BinOpExpr: /
                LiteralExpr: 2
                LiteralExpr: 6
